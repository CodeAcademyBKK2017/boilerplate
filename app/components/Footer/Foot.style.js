import React, {StyleSheet} from 'react-native';

export default StyleSheet.create({
  footerRow: {
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  footSave: {
    textAlign: 'right'
  },
  fonts: {
    fontWeight: '600'
  },
  Foot: {
    marginTop: 20,
    marginBottom: 20
  }
  
});