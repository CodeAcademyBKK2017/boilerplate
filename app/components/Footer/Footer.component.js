import noop from 'lodash/noop';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Footer.style';
import Touchable from 'react-native-platform-touchable';
import {
  Text,
  View
} from 'react-native';

export default class Footer extends Component {
  render () {
    return (
      <View style={styles.container}>
        <View style={styles.status}>
          <Touchable
            style={styles.saveButtonTouch}
            background={Touchable.Ripple('blue')}
            onPress={this.props.onSaveButtonPress}>
            <Text style={styles.saveButtonContent}>Save</Text>
          </Touchable>
          <Touchable
            style={styles.saveButtonTouch}
            background={Touchable.Ripple('blue')}
            onPress={this.props.onShowAboutUs}>
            <Text style={styles.saveButtonContent}>About</Text>
          </Touchable>
          <Text>{this.props.textContentLength} characters</Text>
        </View>
      </View>
    );
  }
}

Footer.propTypes = {
  textContentLength: PropTypes.number.isRequired,
  onSaveButtonPress: PropTypes.func.isRequired,
  onShowAboutUs: PropTypes.func.isRequired
};

Footer.defaultProps = {
  textContentLength: 0,
  onSaveButtonPress: noop,
  onShowAboutUs: noop
};
