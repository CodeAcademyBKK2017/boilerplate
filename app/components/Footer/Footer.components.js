import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './Foot.style';
import {
  Text,
  TouchableOpacity,
  View
} from 'react-native';

class Footer extends Component {
  
  render () {
    return (
      <View style={styles.Foot}>
        <View style={styles.footerRow}>
          <TouchableOpacity  onPress={this.props.noteTitles}>
            <Text style={styles.fonts}>Save</Text>
          </TouchableOpacity>
          <Text style={styles.footSave}>{this.props.texts.length} chacters</Text>
        </View>
        
      </View>
    );
  }
}

Footer.propTypes = {
  texts: PropTypes.string.isRequired,
  noteTitles: PropTypes.func.isRequired
};
export default Footer;