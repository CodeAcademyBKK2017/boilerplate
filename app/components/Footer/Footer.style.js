import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  text: {
    fontSize: 15
  },
  textBold: {
    fontWeight: 'bold',
    fontSize: 15
  },
  container: {
    paddingVertical: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    width: '100%'
  }
  
});