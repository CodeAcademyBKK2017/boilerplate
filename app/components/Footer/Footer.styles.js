import React, {StyleSheet} from 'react-native';

export default StyleSheet.create({
  container: {
    width: '100%',
    flexDirection: 'row',
    justifyContent: 'space-between'
    
  },
  saveButton: {
    fontWeight: 'bold',
    fontSize: 13
  },
  charCount: {
    fontSize: 13
  }
});