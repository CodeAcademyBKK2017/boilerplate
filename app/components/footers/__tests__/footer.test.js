import 'react-native';
import Footer from '../footer.component';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
// import {shallow} from 'enzyme';

describe('Footer', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <Footer />
    );
    expect(tree).toBeDefined();
  });
  it('Footer: renders correctly', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  
});
