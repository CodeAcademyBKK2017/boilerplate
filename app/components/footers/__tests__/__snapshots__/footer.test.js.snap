// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Footer: renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 10,
      "width": "100%",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontWeight": "bold",
          "textAlign": "left",
        }
      }
    >
      Save
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#333333",
        "textAlign": "right",
      }
    }
  >
    150 charector
  </Text>
</View>
`;
